### PR TITLE
Allow Tox to run a single unit test

### PR DESCRIPTION
**Pull request by webbnh**
Currently, one can control what unit tests get run under Tox via a command like the following:
```
tox -- agent python pbench.test.unit.agent.task.test_copy_result_tb
```
Ostensibly, this should run only the tests in the `test_copy_result_tb` module.  Sadly, it does not only that but it also runs 50 or even 500 other tests from `pbench.test.unit.common`, `pbench.test.unit.agent` and `pbench.test.functional.agent`.

This change allows `exec-tests` to skip appending the extra paths to the test list.

However, I have a question for reviewers (i.e., @portante).  The last argument to `tox`/`exec-tests` is named "posargs" and the code comment claims that it is for "positional arguments".  However, the value is inserted after the PyTest `--pyargs` switch, which makes me think that the only acceptable arguments are Python packages.  Is that indeed the case?  We're not supposed to be able to use this mechanism for `pytest` switches?  If so, I can make this change _a lot_ simpler....

(BTW, this change will be even easier to review if you let GitHub hide whitespace changes under the "gear" pull-down at the top of the diff's page.)

### CODE REVIEW

**Comment by portante** [comment]
If you read `tox.ini`, the `posargs` are the set of arguments that `tox` adds after its interpretation of the command line.  They are intended to be any set of arguments applied to the pytest command.

**Review by portante — APPROVED** [review]

**Review by dbutenhof — APPROVED** [review]
Ha; I have a pending comment from days ago that I apparently failed to submit. While the negative logic bugs me a bit, I won't stand in the way after you have an approval. 😬

File: exec-tests
Comment (webbnh) [reply exec-tests:145]: I actually went back and forth when I was writing this.  I settled on the above approach because of the _next_ line (147):
```
    if [[ ${no_tests_specified} -eq 1 ]]; then
```

Does this read better?
```
    tests_specified=0
    for word in ${posargs}; do
        if [[ ${word} != -* ]]; then tests_specified=1; fi
    done
    if [[ ${tests_specified} -eq 0 ]]; then
```
The normal case is that there are no tests specified, so I set up the code to make that be the "true" value.  (I think it would read better without the `-eq X`, but I'm not sure how to do that in Bash....)

Comment (dbutenhof) [reply exec-tests:145]: Yeah, I agree nothing in bash is ideal, but I like `tests_specified` better than `no_tests_specified`. Is it worth another commit and review pass? Well, I dunno.

**Comment by webbnh** [comment]
> If you read `tox.ini`, the `posargs` are the set of arguments that `tox` adds after its interpretation of the command line. They are intended to be any set of arguments applied to the pytest command.

Understood, but if you look at where `${posargs}` is positioned on the `pytest` command line in `exec-tests`, it follows `--pyargs` which, if I read the [docs](https://docs.pytest.org/en/stable/reference/reference.html#command-line-flags) right, affects _all_ subsequent arguments.

So, I think `${posargs}` is impaled on the horns of a dilemma:  if it's supposed to contain one or more tests to run, then it's good; however, if it's supposed to contain PyTest options, then I think it needs to be positioned before the `--pyargs` option...and it cannot be in _both_ places....

File: exec-tests
Comment (portante) [reply exec-tests:145]: According to https://docs.pytest.org/en/7.1.x/example/pythoncollection.html#interpreting-cmdline-arguments-as-python-packages, `--pyargs` is not for options but for arguments, what tests to run and how to interpret those arguments for test discovery.  We can move `--pyargs` into the `pytest.ini` file if that helps.

Comment (webbnh) [reply exec-tests:145]: OK, so long as `--pyargs` is a stand-alone option (i.e., it affects how PyTest interprets command line arguments (and not options)) and doesn't take any argument-value itself, then there's no dilemma.  😅 

Moving `--pyargs` to `pytest.ini` sounds like a good idea, but I don't think I'll add that to this PR.

And, I guess I'll merge this as is, then.  (We can always change it later....)

**Comment by portante** [comment]
This is a very helpful improvement for developers wanting to re-run tests easily.  Thanks!

**Pull request merged**
